### PR TITLE
Support pod conditions for pending job count calculation

### LINKED ISSUE #1963: Support Pod conditions for calculating PendingJobCount in accurate scaling startegy
### Proposal

Currently KEDA looks for the Pod phase (when scaling strategy is set to `accurate`) to determine whether a pod is pending or not in order to avoid excess triggers when pod is waiting to start or create. This feature was introduced in #1391.

I believe this could be enhanced by not only looking at the Pod's phase, but also its conditions. The conditions should be configurable like:
```yml
scalingStategy:
  strategy: "accurate"
  customPendingPodConditions: 
    - "Ready"
    - "PodScheduled"
    - "SomeOtherCustomConditionDefinedByReadinessGate"
```

Another refactor can be to not look at the Pod's `Running` phase at all since the native `PodScheduled: true` condition means the Pod phase is `Running` so `PodScheduled` can always be set in `customPendingPodConditions` (but I'd leave this to whoever implements it)

### Use-Case

SQS scaler with a polling interval of 5s (needs to react ASAP to new messages).
Sometimes the pod is already marked with a `Running` phase but haven't deleted the message from the queue just yet so KEDA triggers another redundant job in the meantime 😢 

Using Readiness Probes (or custom `readinessGate`) would allow me to mark the pod as "not pending" only after assuring the message is no longer in the queue, helping me to avoid those redundant KEDA triggers 💯 

### Anything else?

Thank you for this great solution. Go KEDA! 🥇

### PR DESCRIPTION
**Pull request by yaronya**
Added the `pendingPodConditions` setting for which users can pass a list of [Pod Conditions](https://kubernetes.io/docs/concepts/workloads/pods/_print/#pod-conditions) which Keda will consider when calculating the pending job count.

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)* - https://github.com/kedacore/keda-docs/pull/496
- [x] Changelog has been updated

Resolves #1963 

Signed-off-by: Yaron Yarimi <yaron.yarimi@env0.com>

### CODE REVIEW

File: pkg/scaling/executor/scale_jobs.go [hunk 1]
```diff
@@ -184,6 +184,34 @@ func (e *scaleExecutor) isAnyPodRunningOrCompleted(j *batchv1.Job) bool {
...
 
+func (e *scaleExecutor) areAllPendingPodConditionsFulfilled(j *batchv1.Job, pendingPodConditions []string) bool {
+	opts := []client.ListOption{
+		client.InNamespace(j.GetNamespace()),
+		client.MatchingLabels(map[string]string{"job-name": j.GetName()}),
+	}
+
+	pods := &corev1.PodList{}
+	err := e.client.List(context.TODO(), pods, opts...)
+
```
Comment (zroubalik) [inline pkg/scaling/executor/scale_jobs.go:195]:
```suggestion
```
Nit

**Review by zroubalik — COMMENTED** [review]
Looking good, just a very minor nit 😅

**Review by TsuyoshiUshio — APPROVED** [review]
Thank you for your contribution. Looks good to me.

**Review by zroubalik — APPROVED** [review]
LGTM

**Comment by yaronya** [comment]
@zroubalik thanks 🙏 
Any estimation on when `2.4.0` will be released?

**Comment by zroubalik** [comment]
@yaronya later this week if everything will fine :)

**Pull request merged**